front/lib/api/assistant/conversation.ts

```diff
@@ -480,10 +454,14 @@ async function batchRenderContentFragment(
         "Unreachable: batchRenderContentFragment must be called with only content fragments"
       );
     }
-    const contentFragment = message.contentFragment;
+
+    const contentFragment = new ContentFragmentResource(
+      ContentFragmentModel,
+      message.contentFragment
+    );
```

Comment:
Can we avoid doing `new ContentFragmentResource` outside of the resource file? Given that this is kind of an exception until we move `MessageModel` to its resource, we can probably go with something like:
```suggestion
    const contentFragment = ContentFragmentResource.fromMessage(message);
```